packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5041,16 +5036,25 @@ async function withController<ReturnValue>(
  * @param config - An incomplete ProviderConfig object.
  * @returns The complete ProviderConfig object.
  */
-function buildProviderConfig(config: Partial<ProviderConfig> = {}) {
+function buildProviderConfig(
+  config: Partial<ProviderConfig> = {},
+): ProviderConfig {
+  if (config.type && config.type !== NetworkType.rpc) {
+    return {
+      ...BUILT_IN_NETWORKS[config.type],
+      // This is redundant with the spread operation below, but this was
+      // required for TypeScript to understand that this property was set to an
+      // Infura type.
+      type: config.type,
+      ...config,
+    };
+  }
   return {
     type: NetworkType.rpc,
     chainId: '1337',
     id: undefined,
     nickname: undefined,
```

Comment:
Nit: While we are changing this function, what do you think about removing these properties? They are optional in the ProviderConfig type, so they aren't doing much here.

Reply:
Good idea; done in b53b83b0